src/com/lftechnology/java_training/jan7/FileInput.java

```diff
@@ -0,0 +1,153 @@
+	private static final String fileName = "in.txt";
+
+	public static void main(String[] args) {
+		// read values from file
+		boolean readStatus = readFromFile();
+		if (readStatus) {
+			// display values written in file to console
+			displayValueFromFile();
+		}
+	}
+
+	/**
+	 * read data from file and store it according to its type
+	 */
+	private static boolean readFromFile() {
+		Scanner inputFile = null;
+		try {
+			inputFile = new Scanner(new BufferedReader(new FileReader(fileName)));
+			// loop until the value finishes
+			while (inputFile.hasNext()) {
+				// check if value is int
+				if (inputFile.hasNextInt()) {
+					intReadValue[intValueCounter++] = Integer.parseInt(inputFile.next());
+					// check if value is double
+				} else if (inputFile.hasNextDouble()) {
+					doubleReadValue[doubleValueCounter++] = Double.parseDouble(inputFile.next());
+				} else {
+					stringReadValue[stringValueCounter++] = inputFile.next();
+
+				}
+			}
+			return true;
+		} catch (Exception e) {
+			LOGGER.log(Level.SEVERE, "file not found {0}", e);
+			return false;
+
+		} finally {
+			inputFile.close();
+		}
+
+	}
+
+	/**
+	 * display value stored in different variable which are according to data types
+	 */
+	private static void displayValueFromFile() {
+		String intValue = displayIntegerValue();
+		String doubleValue = displayDoubleValue();
+		String stringValue = displayStringValue();
+		String displaySum = displaySumOfNumbers();
+
+		LOGGER.log(
+				Level.INFO,
+				"\na.The integer read is {0} \nb.The doubling point number read is {1} \nc.the string read is {2} \nd.Hi! {2} The sum of {0}  and {1}  is {3} ",
+				new Object[] { intValue, doubleValue, stringValue, displaySum });
+
+	}
+
+	/**
+	 * return stored value from {@link Integer} array taken from file
+	 * 
+	 * @return all values from {@link Integer} concatinated in a {@link String} separated by comma
+	 */
+	private static String displayIntegerValue() {
+		int tempCounter = intValueCounter;
+		StringBuilder tempIntValues = new StringBuilder();
+		do {
+			tempIntValues.append(intReadValue[--tempCounter]);
+			if (tempCounter != 0) {
+				tempIntValues.append(",");
+			}
+		} while (tempCounter != 0);
+		// changed from StringBuilder to String
+		return tempIntValues + "";
+	}
+
+	/**
+	 * return stored value from {@link Double} array taken from file
+	 * 
+	 * @return all values from {@link Double} concatinated in a {@link String} separated by comma
+	 */
+	private static String displayDoubleValue() {
+		int tempCounter = doubleValueCounter;
+		StringBuilder tempDoubleValues = new StringBuilder();
+		do {
+			tempDoubleValues.append(doubleReadValue[--tempCounter]);
+			if (tempCounter != 0) {
+				tempDoubleValues.append(",");
+			}
+		} while (tempCounter != 0);
+		// changed from StringBuilder to String
+		return tempDoubleValues + "";
+	}
+
+	/**
+	 * return stored value from {@link String} array taken from file
```

Comment:
Not the best way to define the 3 methods separately. Even the defined methods are not proper.